PowerEditor/src/WinControls/FileBrowser/fileBrowser.h

```diff
@@ -176,6 +176,8 @@ class FileBrowser : public DockingDlgInterface {
 	HMENU _hFileMenu = NULL;
 	std::vector<FolderUpdater *> _folderUpdaters;
 
+	generic_string _fullPath;

```

Comment:
Could you change this line to the following:
```
generic_string _selectedNodeFullPath; // this member is used only for PostMessage call
```

Reply:
> My question is why PostMessage makes the difference?

Short answer: Because the **TreeView** double click handler claims back the input focus after calling the double click notification procedure.

Long answer: You cannot debug it with Spy++, but by including `OutputDebugString` calls into the source code of Npp. If you write a debug message starting with '++' before certain calls and a 2nd debug message starting with '--' after the same calls, then it gives the following picture:

	SendMessage
	===========
	001 ++TreeView WM_LBUTTONDBLCLK          <===== POSTED
	002   ++ProjectPanel::notified(NM_DBLCLK)
	003     ++SendMessage(_hParent, NPPM_DOOPEN)
	004       ++Notepad_plus::process(NPPM_DOOPEN)
	005         TreeView WM_KILLFOCUS
	006         ScintillaEditView WM_SETFOCUS
	007       --Notepad_plus::process(NPPM_DOOPEN)
	008     --SendMessage(_hParent, NPPM_DOOPEN)
	009   --ProjectPanel::notified(NM_DBLCLK)
	010   ScintillaEditView WM_KILLFOCUS
	011   TreeView WM_SETFOCUS
	012 --TreeView WM_LBUTTONDBLCLK

	PostMessage
	===========
	101 ++TreeView WM_LBUTTONDBLCLK          <===== POSTED
	102   ++ProjectPanel::notified(NM_DBLCLK)
	103     ++PostMessage(_hParent, NPPM_DOOPEN)
	104     --PostMessage(_hParent, NPPM_DOOPEN)
	105   --ProjectPanel::notified(NM_DBLCLK)
	106 --TreeView WM_LBUTTONDBLCLK
	107 ++Notepad_plus::process(NPPM_DOOPEN) <===== POSTED
	108   TreeView WM_KILLFOCUS
	109   ScintillaEditView WM_SETFOCUS
	110 --Notepad_plus::process(NPPM_DOOPEN)

First shown is the sequence of events using `SendMessage`. The `NPPM_DOOPEN` command is processed during the `SendMessage` call, the input focus is transferred to Scintilla in lines `005` and `006`, and is claimed back by  **TreeView** in lines `010` and `011`, before the processing of the double click message ends.

The second sequence shows what happens using `PostMessage`. `PostMessage` returns immediately, leaving the `NPPM_DOOPEN` command in the message queue of Npp. The processing of the double click message by **TreeView** ends without focus fighting, and after that, the `NPPM_DOOPEN` command is processed as intended. The input focus switching at lines `108` and `109` cannot be contested anymore by the **TreeView** double click handler.

Important to note is, that only the events marked as POSTED in the sequences shown above come from the message queue. All other actions are plain calls, `SendMessage` performs nothing else than a plain call of the Window procedure of the window specified.

Reply:
@Uhf7 Thank you for the very detail explanation!